src/prefect/tasks/gsheets/gsheets.py

```diff
@@ -0,0 +1,138 @@
+        self.worksheet_name = worksheet_name
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("credentials_filename", "sheet_key", "worksheet_name")
+    def run(
+        self,
+        data: List[Any],
+        credentials_filename: Union[str, pathlib.Path] = None,
+        sheet_key: str = None,
+        worksheet_name: str = None,
+    ) -> dict:
+        """
+        Appends a row of data to a Google Sheets worksheet
+
+        Args:
+            - data (list): the data to insert. This should be formatted as a list
+            - credentials_filename (Union[str, pathlib.Path]): Location of credentials file
+            - sheet_key (str): The key corresponding to the Google Sheet
+            - worksheet_name (str): The worksheet to target
+
+        Returns:
+            - a dictionary containing information about the successful insert
+        """
+        client = gspread.service_account(filename=credentials_filename)
+        google_sheet = client.open_by_key(sheet_key)
+        worksheet = google_sheet.worksheet(worksheet_name)
+        return worksheet.append_row(data)
+
+
+class ReadGsheetRow(Task):
+    """
+    A task for reading a row from a Google Sheet.
+    Note that _all_ initialization settings can be provided / overwritten at runtime.
+
+    Args:
+        - credentials_filename (Union[str, pathlib.Path]): Location of credentials file
+        - sheet_key (str): The key corresponding to the Google Sheet
+        - worksheet_name (str): The worksheet to target
+        - **kwargs (optional): additional kwargs to pass to the `Task` constructor
+    """
+
+    def __init__(
+        self,
+        credentials_filename: Union[str, pathlib.Path] = None,
+        sheet_key: str = None,
+        worksheet_name: str = None,
+        **kwargs: Any
+    ):
+        self.credentials_filename = credentials_filename
+        self.sheet_key = sheet_key
+        self.worksheet_name = worksheet_name
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("credentials_filename", "sheet_key", "worksheet_name")
+    def run(
+        self,
+        row: int,
+        credentials_filename: Union[str, pathlib.Path] = None,
+        sheet_key: str = None,
+        worksheet_name: str = None,
+    ) -> List[Any]:
+        """
+        Appends a row of data to a Google Sheets worksheet
+
+        Args:
+            - row (int): The number of the row to read
+            - credentials_filename (Union[str, pathlib.Path]): Location of credentials file
+            - sheet_key (str): The key corresponding to the Google Sheet
+            - worksheet_name (str): The worksheet to target
+
+        Returns:
+            - a list of values from the row 
+        """
+        client = gspread.service_account(filename=credentials_filename)
+        google_sheet = client.open_by_key(sheet_key)
+        worksheet = google_sheet.worksheet(worksheet_name)
+        return worksheet.row_values(row)
+
+
+def gsheet_helper(fn: Callable):

```

Comment:
Is this helper necessary? It creates some helper tasks, but instills certain signature decisions on them that others may not want. Since the content of `inner` is rather short, I wonder if leaving these up to users would be better? No strong thoughts.

Reply:
The `ghsheet_helper` was in the outline.toml but removed because it was breaking the doc generation https://app.circleci.com/pipelines/github/PrefectHQ/prefect/4487/workflows/9f7287d9-2140-4551-8521-a96028da3868/jobs/83420/steps

Reply:
Right on!  I figured it'd be useful to to have a "wrapper" that let you turn anything you can do with the `gsheets` package into a Prefect Task (sorta like having a generic "SQL Query" task), but I'm not sure if they way I did it interacts well with the rest of the ecosystem.

Reply:
My gut says we should leave this out - users can write their own `@task` decorated tasks for handling this kind of thing, and may want more flexibility in task format than this provides. For example, they might want to manage clients differently, rather than creating a new one per task.

I also don't find:

```python
find_dough = gsheet_helper(lambda x: x.find("Dough"))

res = find_dough(credentials_filename, sheet_key, worksheet_name)
```

much more readable than

```python
@task
def find_dough(credentials_filename, sheet_key, worksheet_name):
    client = gspread.service_account(filename=credentials_filename)
    google_sheet = client.open_by_key(sheet_key)
    worksheet = google_sheet.worksheet(worksheet_name)
    return worksheet.find("Dough")

res = find_dough(credentials_filename, sheet_key, worksheet_name)
```

just my opinion though. cc @cicdw, @joshmeek for their thoughts.

Reply:
Besides this thread here, the rest of this PR looks good to me. Thanks for all your work here @mattalhonte!

Reply:
Yea I vote remove it - in general I think a "task factory" like this shouldn't ever be necessary because Prefect Tasks can be configured through both init kwargs + runtime data

Reply:
Roger!